create_sig.py

```diff
@@ -0,0 +1,228 @@
+    # on Windows need to shorten file names to avoid command line >~ 8100 characters
+    created_temp_files = False
+    if os.name == "nt":
+        logger.debug("creating temporary pat files")
+        pat_files = create_temp_pat_files(pat_files)
+        created_temp_files = True
+
+    args = [PATH_SIGMAKE, "-vvv"] + pat_files + [outsigfile]
+    logfile = open("run1.log", "wb")
+    run(args, stdout=logfile, stderr=logfile)
+    logfile.close()
+
+    exc_file = os.path.splitext(outsigfile)[0] + ".exc"
+    process_exc_file(exc_file)
+
+    logfile = open("run2.log", "wb")
+    run(args, stdout=logfile, stderr=logfile)
+    logfile.close()
+
+    if created_temp_files:
+        remove_files(pat_files)
+
+    os.chdir(curdir_old)
+
+    if not os.path.exists(outsigfile):
+        raise ValueError(f"did not create {outsigfile}")
+
+    return outsigfile
+
+
+def create_temp_pat_files(files):
+    temp_pat_files = []
+    for i, f in enumerate(files):
+        new = f"{i}.pat"
+        shutil.copy(f, new)
+        temp_pat_files.append(new)
+    return temp_pat_files
+
+
+def remove_files(pat_files):
+    for f in pat_files:
+        logger.debug("removing file %s", f)
+        os.remove(f)
+
+
+def get_pat_files(path):
+    pat_paths = []
+    for root, dirs, files in os.walk(path):
+        for file in files:
+            if file.endswith(".pat"):
+                pat_path = os.path.join(root, file)
+                pat_paths.append(os.path.relpath(pat_path))
+    return pat_paths
+
+
+def run(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE):
+    logger.debug("running %s", " ".join(args))
+    p = subprocess.Popen(args, stdout=stdout, stderr=stderr)
+    # wait
+    p.communicate()
+
+
+def process_exc_file(exc_file):
+    """
+    for now just delete first 5 lines
+    """
+    with open(exc_file, "rb") as f:
+        d = f.read().splitlines(True)
+    with open(exc_file, "wb") as f:
+        f.writelines(d[5:])
+
+
+def zipsig(outsigfile):
+    logger.debug(outsigfile)
+    shutil.copy(outsigfile, f"{outsigfile}.bu")
+
+    args = [PATH_ZIPSIG, outsigfile]
+    run(args)
+
+
+def main(argv=None):
+    if argv is None:
+        argv = sys.argv[1:]
+
+    desc = "create a .sig file based on existing pat files stored in tarballs"
+    parser = argparse.ArgumentParser(description=desc)
+
+    parser.add_argument(
+        "path",
+        type=str,
+        help="data path to root of tarballs",
+    )
+    parser.add_argument(
+        "outdir",
+        type=str,
+        help="path to output directory",
+    )
+    parser.add_argument("-e", "--exclude", nargs="*", help="exclude paths that contain exclusion string")
+    parser.add_argument("-i", "--include_pats", nargs="*", help="include pat files that contain inclusion string")
+    parser.add_argument("-d", "--debug", action="store_true", help="enable debugging output on STDERR")
+    parser.add_argument("-s", "--sigfile", default="flare.sig", help="name of output sigfile")
```

Comment:
lets add `--patfile` so its easy to recover the source data used to generate the `.sig`?

Reply:
Do you mean to explicitly specify `--patfile ./data/VS9/VC/lib/libcmt.lib.pat ./data/VS9/VC/lib/libcmtd.lib.pat ...`?

Reply:
added in 3eb6ffb, providing two ways now to specify .pat files